lib/OpenQA/WebSockets/Server.pm

```diff
@@ -250,12 +259,19 @@ sub _message {
                     return unless $w;
                     log_debug("Updated worker seen from worker_status");
                     $w->seen;
+                    $w->set_property('VERSION', $worker_version);
                 });
         }
         catch {
             log_error("Failed updating worker seen status: $_");
         };
 
+        if ($WORKERS_REQUIRED_VERSION > $worker_version) {
+            log_debug("Received a message from an incompatible worker");
+            $ws->tx->send({json => {type => 'incompatible'}});
```

Comment:
I would also close the connection in such case, like 
```perl
$ws->finish("1008", "Connection terminated from WebSocket server - incompatible communication protocol version");
``` 
or such (note: there are also  specific  test cases for that).

Reply:
I have updated the code to shutdown the connection, in case the worker is too old to understand the "incompatible" message type.